core/src/main/java/jenkins/MissingDependencyException.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import hudson.PluginWrapper.Dependency;
+import hudson.Util;
+
+/**
+ * Exception thrown if plugin resolution fails due to missing dependencies
+ * 
+ * @author Carlos Sanchez
+ * @since 1.650
+ *
+ */
+public class MissingDependencyException extends IOException {
```

Comment:
I don't disagree that it should be a different exception type, just saying that if somebody is now catching IOException to handle dependency problems and we change it to something else it will break. 
If that's not an issue I'll create a jira to account for it in the changelog